src/core/geometry/qgsgeometrycollection.cpp

```diff
@@ -36,7 +36,8 @@ QgsGeometryCollection::QgsGeometryCollection()
   mWkbType = QgsWkbTypes::GeometryCollection;
 }
 
-QgsGeometryCollection::QgsGeometryCollection( const QgsGeometryCollection &c ): QgsAbstractGeometry( c )
+QgsGeometryCollection::QgsGeometryCollection( const QgsGeometryCollection &c ):
+  QgsAbstractGeometry( c ), mHasCachedValidity( false ), mValidityFailureReason()
```

Comment:
we could copy the cached validity flag/reason here too

Reply:
makes sense. Done